src/libDirectoryService/FinalBlockPreProcessing.cpp

```diff
@@ -449,13 +449,15 @@ bool DirectoryService::RunConsensusOnFinalBlockWhenDSPrimary()
     // finalBlockMessage = serialized final block + tx-body sharing setup
     vector<unsigned char> finalBlockMessage = ComposeFinalBlockMessage();
 
-    // kill first ds leader
-    //if (m_consensusMyID == 0 && temp_todie)
-    //{
-    //    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-    //                 "I am killing myself to test view change");
-    //    throw exception();
-    //}
+    // kill first ds leader (used for view change testing)
```

Comment:
Why need to preserve this?

Reply:
Although ugly, we got to preserve it for future view change testing. I can't think of a simple unit test to remove this part.